pyanaconda/modules/storage/partitioning/automatic_interface.py

```diff
@@ -0,0 +1,186 @@
+#
+# DBus interface for the auto partitioning module.
+#
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.dbus.interface import dbus_interface
+from pyanaconda.dbus.property import emits_properties_changed
+from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
+from pyanaconda.modules.common.base import KickstartModuleInterfaceTemplate
+from pyanaconda.modules.common.constants.objects import AUTO_PARTITIONING
+from pyanaconda.modules.storage.constants import AutoPartitioningType
+
+
+@dbus_interface(AUTO_PARTITIONING.interface_name)
+class AutoPartitioningInterface(KickstartModuleInterfaceTemplate):
+    """DBus interface for the auto partitioning module."""
+
+    def connect_signals(self):
+        """Connect the signals."""
+        super().connect_signals()
+
+        self.watch_property("Enabled", self.implementation.enabled_changed)
+        self.watch_property("Type", self.implementation.type_changed)
+        self.watch_property("FilesystemType", self.implementation.fstype_changed)
+        self.watch_property("NoHome", self.implementation.nohome_changed)
+        self.watch_property("NoBoot", self.implementation.noboot_changed)
+        self.watch_property("NoSwap", self.implementation.noswap_changed)
+        self.watch_property("Encrypted", self.implementation.encrypted_changed)
+        self.watch_property("Cipher", self.implementation.cipher_changed)
+        self.watch_property("Passphrase", self.implementation.passphrase_changed)
+        self.watch_property("Escrowcert", self.implementation.escrowcert_changed)
+
+        self.watch_property(
+            "BackupPassphraseEnabled", self.implementation.backup_passphrase_enabled_changed
+        )
+
+    @property
+    def Enabled(self) -> Bool:
+        """Is the auto partitioning enabled?"""
+        return self.implementation.enabled
+
+    @emits_properties_changed
+    def SetEnabled(self, enabled: Bool):
+        """Is the auto partitioning enabled?
+
+        :param enabled: a boolean value
```

Comment:
I think this can go away (or tell about semantics, not type).

Reply:
Fixed.